### DISCUSSION TITLE
How correctly intercept and resolve $fetch response, when there is an error?

### DISCUSSION THREAD

**Discussion by engenes** [nuxt/nuxt#17108] [Questions]
I want to intercept "not found error" and skip others
I make this:

**Call function in component**

 ```
const fetchOnClick = async () => {
  const data = await $fetch('bad_url', {
    onResponseError(ctx) {
      if (ctx.response.status === 404) {
        console.warn('Not found');

        return {
          status: 'error',
          error: {
            message: 'not found',
          },
        };
      }
    },
  });

  console.log(data);
};
```
After error response I see `Not found`,  but `data` does not appear in log
That is, with such an error, I still want to get the correct message and clear the error.
So that only data that does not pass the condition enters the error

**Comment by Flanker72** [comment]
Seems you can create own wrapper over the `$fetch` using `$fetch.raw` - you can handle errors and control the return value.
